modin/pandas/__init__.py

```diff
@@ -106,16 +106,19 @@ def _update_engine(publisher: Parameter):
     from modin.config import Backend, CpuCount
 
     if publisher.get() == "Ray":
-        from modin.engines.ray.utils import initialize_ray
+        if _is_first_update.get("Ray", True):
+            from modin.engines.ray.utils import initialize_ray
 
+            initialize_ray()
+    elif publisher.get() == "Native":
         # With OmniSci backend there is only a single worker per node
         # and we allow it to work on all cores.
         if Backend.get() == "Omnisci":
-            CpuCount.put(1)
             os.environ["OMP_NUM_THREADS"] = str(multiprocessing.cpu_count())
```

Comment:
I wonder if we should actually use `CpuCount.get()` here instead to allow the user to control this

Reply:
Good point! Should be corrected.

Reply:
Updated to use `CpuCount.get()`